### ISSUE TITLE
Feature Request: global file

### ISSUE THREAD

**Issue by m4ss1m0g** [xaviersolau/BlazorJsonLocalization#39]
## Request
It is possible to have a global file as a fallback ?

## Reason
Many times there are _global_ messages like **Cancel** or **Open** that are much better to have on a central file, instead of repeating them on all files.

## Implementation 
When I wrote `@Loc["lblName"]` the framework must do:

1. Search in the specific file xxx (same as today)
2. IF NOT FOUND on the specific file, look in the root folder Resources for a file like global-es-US.json (maybe the name/path can be customized)

**Comment by xaviersolau** [comment]
Thanks for your request.

Did you consider injecting a Global localizer in addition of your specific one:

```csharp
[Inject]
private IStringLocalizer<MyComponent> Loc { get; set; }

[Inject]
private IStringLocalizer<Global> GlobalLoc { get; set; }
```

**Comment by m4ss1m0g** [comment]
The docs say:

>For example if your Index page is in the Pages sub-folder, your Index Json files must be in the same sub-folder: Resources/Pages >with Resources the resources path defined in the options.

So in this case I must have a `Global-xx-XX.json`  per each component, and even if I could specify a Root component I always repeat the import of Global json. I'm porting a Vue app where I use [Vuei18n](https://kazupon.github.io/vue-i18n/guide/interpolation.html#basic-usage) the usage is pretty similar the installation is quite different, for example on Vuei18n you start with a global translation file(s) and you can add more translations for a specific component.

I think this approach has the convenience to start with a root folder where add your localization  without specifying the folder structure, then if the project grows you can add a specific translation for a specific component.

**Comment by xaviersolau** [comment]
Well actually you can find an example in the repository in the project `SoloX.BlazorJsonLocalisation.Example.Components.Embedded`.

In the example, you can find a shared resource file in the root folder and a specific Index resource file. It is very similar to the use case you are describing:

![image](https://user-images.githubusercontent.com/15972064/187775116-3b84dc97-a620-445d-bbbc-1f7a999759ca.png)

In order to be able to load the resources you can see that there is a class named `SharedResource` (witch is empty because we just need it when you inject the shared localizer).

```csharp
[Inject]
private IStringLocalizer<Index> IndexLoc { get; set; }
[Inject]
private IStringLocalizer<SharedResource> SharedLoc { get; set; }
```

Does this example respond to your need?
Or do you really want to access the shared resources transparently? Without using the injected SharedLoc property?

From my point of view I think using explicitly the injected property is a little more readable....

**Comment by m4ss1m0g** [comment]
In this scenario, you need to add the `IStringLocalizer<SharedResource>` and an empty .cs file, and **from my point of view** add more _entropy_.
With a shared and transparent global file, I think you have the advantage of zero configs, and no subfolders to create/configure, maybe for a small project and for limited resources is a better choice.

My Opinion.

**Comment by xaviersolau** [comment]
Ok I think we can find a intermediary solution.

Let's say that:
* we have this SharedResource class (or interface)
* we declare in the config that we use a fallback on this class
* or Index (in our example) inherit the SharedResource? (Why not?)

then we wouldn't need to add the IStringLocalizer<SharedResource> property.

What do you think?

**Comment by m4ss1m0g** [comment]
A good solution.

I look at the code (very well written by the way), and another solution could be to add on your lib a fix file SharedResources.cs and modify the [Helpers](https://github.com/xaviersolau/BlazorJsonLocalization/blob/main/src/libs/SoloX.BlazorJsonLocalization/Helpers/ResourcePathHelper.cs#L17) to replace your namespace with the app namespace when you look for the SharedResource.cs file.

I understand that the feature request is not straightforward because the lib is based on the Assembly parameter on pretty much all functions if I understand correctly.

I suggest this modification (and the structured JSON on issue #32) because I think this lib is pretty much the same as Vue18n and I think that when people come from vue.js to Blazor they are looking for the same Vue18n behaviors and I hope this lib will be ready.

**Comment by xaviersolau** [comment]
Working on (#43)

**Comment by xaviersolau** [comment]
This is available in version 2.0.0-alpha.3

**Comment by m4ss1m0g** [comment]
@xaviersolau Thanks for implementing the features, the global file solution is very nice!
